### PR TITLE
[Serializer] [5.0] Add type-hints to serializer interface

### PR DESCRIPTION
**Pull request by Simperfit**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | no 
| BC breaks?    | no     
| Deprecations? | no 
| Tests pass?   | yes    
| Fixed tickets | #32179   
| License       | MIT
| Doc PR        | none 

This PR adds type-hints to the serializer component ;).

### CODE REVIEW

**Review by dunglas — APPROVED** [review]

File: src/Symfony/Component/Serializer/Encoder/DecoderInterface.php [hunk 1]
```diff
@@ -36,7 +36,7 @@ interface DecoderInterface
      *
      * @throws UnexpectedValueException
      */
-    public function decode($data, $format, array $context = []);
+    public function decode($data, string $format, array $context = []);
```
Comment (derrabus) [inline src/Symfony/Component/Serializer/Encoder/DecoderInterface.php]: Either we can type-hint `string $data` or the docblock is a lie. 🤔

Comment (Simperfit) [reply src/Symfony/Component/Serializer/Encoder/DecoderInterface.php]: done :)

**Review by derrabus — APPROVED** [review]

File: src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php [hunk 2]
```diff
@@ -121,15 +121,15 @@ public function __construct(ClassMetadataFactoryInterface $classMetadataFactory
     /**
      * {@inheritdoc}
      */
-    public function supportsNormalization($data, $format = null)
+    public function supportsNormalization($data, ?string $format = null)
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php]: please remove all `?` to all arguments that default to null, for consistency with the CS

Comment (Simperfit) [reply src/Symfony/Component/Serializer/Normalizer/AbstractObjectNormalizer.php]: done

**Review by fabpot — APPROVED** [review]

**Comment by fabpot** [comment]
Thank you @Simperfit.

**Pull request merged**